public/pages/HistoricalDetectorList/containers/HistoricalDetectorList.tsx

```diff
@@ -0,0 +1,325 @@
+import { HistoricalDetectorListItem } from '../../../models/interfaces';
+import { SORT_DIRECTION } from '../../../../server/utils/constants';
+import ContentPanel from '../../../components/ContentPanel/ContentPanel';
+import { AppState } from '../../../redux/reducers';
+import { getHistoricalDetectorList } from '../../../redux/reducers/ad';
+import {
+  BREADCRUMBS,
+  APP_PATH,
+  PLUGIN_NAME,
+  MAX_DETECTORS,
+} from '../../../utils/constants';
+import { DETECTOR_STATE } from '../../../../server/utils/constants';
+import { getTitleWithCount } from '../../../utils/utils';
+import { ALL_DETECTOR_STATES } from '../../utils/constants';
+import { getURLQueryParams } from '../utils/helpers';
+import {
+  filterAndSortHistoricalDetectors,
+  getHistoricalDetectorsToDisplay,
+} from '../../utils/helpers';
+import { EmptyHistoricalDetectorMessage } from '../components/EmptyHistoricalDetectorMessage/EmptyHistoricalDetectorMessage';
+import { historicalDetectorListColumns } from '../utils/constants';
+import { HistoricalDetectorFilters } from '../components/HistoricalDetectorFilters/HistoricalDetectorFilters';
+import { GET_ALL_DETECTORS_QUERY_PARAMS } from '../../utils/constants';
+import { CoreStart } from '../../../../../../src/core/public';
+import { CoreServicesContext } from '../../../components/CoreServices/CoreServices';
+import {
+  NO_PERMISSIONS_KEY_WORD,
+  prettifyErrorMessage,
+} from '../../../../server/utils/helpers';
+
+export interface HistoricalDetectorListRouterParams {
+  from: string;
+  size: string;
+  search: string;
+  sortDirection: SORT_DIRECTION;
+  sortField: string;
+}
+interface HistoricalDetectorListProps
+  extends RouteComponentProps<HistoricalDetectorListRouterParams> {}
+interface HistoricalDetectorListState {
+  page: number;
+  queryParams: any;
+  selectedDetectorStates: DETECTOR_STATE[];
+}
+
+export function HistoricalDetectorList(props: HistoricalDetectorListProps) {
+  const core = React.useContext(CoreServicesContext) as CoreStart;
+  const dispatch = useDispatch();
+
+  // get historical detector store
+  const adState = useSelector((state: AppState) => state.ad);
+  const allDetectors = adState.historicalDetectorList;
+  const errorGettingDetectors = adState.errorMessage;
+  const isRequestingDetectors = adState.requesting;
+
+  const [selectedDetectors, setSelectedDetectors] = useState(
+    [] as HistoricalDetectorListItem[]
+  );
+  const [detectorsToDisplay, setDetectorsToDisplay] = useState(
+    [] as HistoricalDetectorListItem[]
+  );
+  const [isLoadingFinalDetectors, setIsLoadingFinalDetectors] = useState<
+    boolean
+  >(true);
+
+  // if loading detectors or sorting/filtering: set whole page in loading state
+  const isLoading = isRequestingDetectors || isLoadingFinalDetectors;
+
+  const [state, setState] = useState<HistoricalDetectorListState>({
+    page: 0,
+    queryParams: getURLQueryParams(props.location),
+    selectedDetectorStates: ALL_DETECTOR_STATES,
```

Comment:
Do we use the same list of detector states for realtime and historical detectors?

Reply:
Yes, but with some exclusive to real-time (`FEATURE_REQUIRED`, `INIT_FAILURE`) and some exclusive to historical (`FINISHED`) that are filtered out on the appropriate pages. The reason I didn't separate is because of the long-term goal of having detection tasks for both real-time and historical detectors.

Reply:
ok, sounds good